torchvision/transforms/functional_tensor.py

```diff
@@ -236,3 +276,51 @@ def _blend(img1, img2, ratio):
     # type: (Tensor, Tensor, float) -> Tensor
     bound = 1 if img1.dtype in [torch.half, torch.float32, torch.float64] else 255
     return (ratio * img1 + (1 - ratio) * img2).clamp(0, bound).to(img1.dtype)
+
+
+def _rgb2hsv(img):
+    r, g, b = img[0], img[1], img[2]
+
+    maxc, _ = torch.max(img, dim=0)
+    minc, _ = torch.min(img, dim=0)
+    uv = maxc
+
+    cr = maxc.to(dtype=torch.float32) - minc.to(dtype=torch.float32)
+    s = cr / maxc.to(dtype=torch.float32)
+    rc = (maxc-r).to(dtype=torch.float32) / cr
+    gc = (maxc-g).to(dtype=torch.float32) / cr
+    bc = (maxc-b).to(dtype=torch.float32) / cr
+
+    s = (maxc != minc) * s
+    hr = (maxc == r) * (bc - gc)
+    hg = ((maxc == g) & (maxc != r)) * (2.0 + rc - bc)
+    hb = ((maxc != g) & (maxc != r)) * (4.0 + gc - rc)
+    h = (hr + hg + hb)
+
+    #  torch.fmod and  math.fmod have different precision.
+    h = torch.fmod((h / 6.0 + 1.0), 1.0)
+
+    if img.dtype == torch.uint8:
+        uh = torch.clamp((h*255.0).to(dtype=torch.int32), 0, 255).to(dtype=torch.uint8)
+        us = torch.clamp((s*255.0).to(dtype=torch.int32), 0, 255).to(dtype=torch.uint8)
+    else:
+        uh = torch.clamp(h * 255.0, 0.0, 255.0)
+        us = torch.clamp(s * 255.0, 0.0, 255.0)
+
+    return torch.stack((uh, us, uv))
+
+def _hsv2rgb(img):
+    h, s, v = img[0], img[1], img[2]
+    i = torch.floor(h.to(dtype=torch.float32) * 6.0 / 255.0).to(dtype=torch.int32)
+    f = (h.to(dtype=torch.float32) * 6.0 / 255.0) - i.to(dtype=torch.float32)
+    fs = s.to(dtype=torch.float32) / 255.0
+
+    p = torch.clamp(torch.round(v.to(dtype=torch.float32) * (1.0 - fs)), 0, 255).to(dtype=img.dtype)
+    q = torch.clamp(torch.round(v.to(dtype=torch.float32) * (1.0 - fs * f)), 0, 255).to(dtype=img.dtype)
+    t = torch.clamp(torch.round(v.to(dtype=torch.float32) * (1.0 - fs * (1.0 - f))), 0, 255).to(dtype=img.dtype)
+    i = i % 6
+
+    r = (i == 0) * v + (i == 1) * q +  (i == 2) * p + (i == 3) * p + (i == 4) * t + (i == 5) * v
+    g = (i == 0) * t + (i == 1) * v +  (i == 2) * v + (i == 3) * q + (i == 4) * p + (i == 5) * p
+    b = (i == 0) * p + (i == 1) * p +  (i == 2) * t + (i == 3) * v + (i == 4) * v + (i == 5) * q
```

Comment:
This is fine, but one could probably batch the equalities as well, something like
```python
mask = i[..., None] == torch.arange(6)
```
which uses broadcasting.
The dimension to broadcast is TBD, could be either the first or the last dimension (maybe the first dimension is better here?)